### PR TITLE
CMake: Removed files used by nothing in the build system

### PR DESCRIPTION
**Pull request by marcusmueller**

### CODE REVIEW

**Review by jdemel — APPROVED** [review]
LGTM. Thanks for this simplification!

At first, I thought about bumping the minimum Boost version. However, Boost is only required on old systems that lack compilers with `std::filesystem` support. A modern system comes with a modern compiler and doesn't require Boost at all.

File: apps/CMakeLists.txt [hunk 1]
```diff
@@ -48,10 +48,7 @@ target_include_directories(volk_profile
 )
 
 if(NOT FILESYSTEM_FOUND)
-    target_include_directories(volk_profile
-        PUBLIC ${Boost_INCLUDE_DIRS}
-    )
-    target_link_libraries(volk_profile PRIVATE ${Boost_LIBRARIES})
+    target_link_libraries(volk_profile PRIVATE Boost::filesystem)
```
Comment (michaelld) [inline apps/CMakeLists.txt:51]: Still need to use the `Boost_INCLUDE_DIRS` for includes here

**Review by michaelld — CHANGES_REQUESTED** [review]
1 required change before I'll accept this. Also a meta-discussion, thinking we want to update minimum Boost to something reasonably recent, like even 1.60.0. When was `Boost::filesystem` added? At issue here is that by keeping min version at 1.35, we still need the `Boost_NOGO_VERSIONS` stuff from `VolkBOOST.cmake`. So, either we leave this all alone or we bump the Boost version so-as to not need the NOGO stuff.

**Comment by marcusmueller** [comment]
Re: boost bumping: yes, I'd say we should do that. 1.35 is positively ancient, pulling up with GNU Radio makes sense to me – chances are that non-GR projects that use VOLK have at least the same Boost versions as GR 3.8 could rely on

File: apps/CMakeLists.txt (hunk 1 above)
Comment (marcusmueller) [reply apps/CMakeLists.txt:51]: `target_link_libraries` with a config target does add the includes to the target, or am I mistaken?

For example, gnuradio/gr-uhd/lib/CMakeLists.txt: https://github.com/gnuradio/gnuradio/blob/master/gr-uhd/lib/CMakeLists.txt#L34

We `target_link_library` the `gnuradio-uhd` target against the UHD::UDH target, which publicly says "hey, include this", which leads to the target automagically gaining these include dependencies.

**Comment by jdemel** [comment]
I'm OK to bump the minimum Boost version. I just don't see the need to do so. Say, we bump the minimum version to 1.65. In that case, we can assume a reasonable recent compiler to be available. This compiler will support `std::filesystem` and thus, we don't need Boost at all.

We removed the last CI case for Linux, when we dropped tests for Ubuntu 14.04.
AppVeyor CI is the only test where we still add Boost. And we can probably remove that too because these tests rely on ancient toolchains.

Long story short. Maybe it's time to have another look at #174 and go through with it.

**Comment by jdemel** [comment]
If we decide to finally purge Boost, we would merge #557 and close #174 .
PR #557 supersedes this PR because it goes even further. However, we need to decide if it is time to purge Boost.

**Comment by michaelld** [comment]
Boost is purged from #557  ... so, closing this PR as it has been superseded!

**Comment by michaelld** [comment]
OK ... will be added ... soon. Closing this is safe :)
